### DISCUSSION TITLE
How do I push coverage reports?

### DISCUSSION THREAD

**Discussion by prologic** [qiniu/goc#311] [Q&A]
Hi 👋

Sorry for the dumb ass question, but I _really_ don't get what `goc` really is. From looking at it's `--help` I get the impression its not the type of coverage reporting service I was after.

For example, I want to do this:

```console
$ go test -cover ./...
$ goc pushlish coverage.out
```

Or something similar... I might be getting the wrong impression of this tool/service...

**Comment by CarlJi** [comment]
`goc` mainly designed to help collect code coverage for the e2e testing during the application's running status. you can try it via [this example ](https://github.com/qiniu/goc#code-coverage-collection-for-your-golang-system-tests)

**Comment by CarlJi** [comment]
while, for your question "push coverage reports", i am thinking you probably means how to use this coverage report, right? usually, differently company has different presentation ways, but maybe [codecov service](https://docs.codecov.com/docs/supported-languages) is also a choice.

**Reply by prologic** [reply]
That's just it. I'm actually trying to find a self-hosted version of a "code coverage" service that I can run on my own infra. I haven't been able to find one, goc is _close_ but as you pointed out, designed for a different puprose.

Thank you for your answer 🙏
